[C#][aspnetcore] Docker build option broken

##### Description

When creating an ASP.NET Core server stub, the Docker build option from `README.md` doesn't work.

##### Swagger-codegen version

`v2.2.2` or latest `master`, doesn't matter.

##### Swagger declaration file content or url

`sh ./bin/aspnetcore-petstore-server.sh` or your own `Swagger.json`, doesn't matter.

##### Steps to reproduce

0. `git clone https://github.com/swagger-api/swagger-codegen && cd swagger-codegen`
1. `git checkout origin master && git pull origin master --rebase`
2. `mvn clean package`
3. `sh ./bin/aspnetcore-petstore-server.sh`
4. `cd samples/server/petstore/aspnetcore/src/IO.Swagger`
5. 
    a. `docker build -t IO.Swagger .`
    b. `docker build -t swagger-petstore .`
6. `docker run -p 5000:5000 swagger-petstore`

##### Issues to fix

There are two issues here:

1. The [README.md](https://github.com/swagger-api/swagger-codegen/blob/c70a2a7481b34362657546e6ca0f4a41ff1c44b3/modules/swagger-codegen/src/main/resources/aspnetcore/README.mustache) template just pulls the [packageName](https://github.com/swagger-api/swagger-codegen/blob/c70a2a7481b34362657546e6ca0f4a41ff1c44b3/modules/swagger-codegen/src/main/resources/aspnetcore/README.mustache#L25) from [AbstractCSharpCodegen](https://github.com/swagger-api/swagger-codegen/blob/c70a2a7481b34362657546e6ca0f4a41ff1c44b3/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractCSharpCodegen.java#L23) (or [whatever you've set in your config](https://github.com/swagger-api/swagger-codegen/blob/c70a2a7481b34362657546e6ca0f4a41ff1c44b3/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AspNetCoreServerCodegen.java#L42)) for the `docker build` example, even though docker only supports lowercase letters for `repo/tag`. Hence, **5a)** will fail on the `Petstore` (`packageName: IO.Swagger`) example and real-world .NET applications that use uppercase letters for the package name. For reference, here's the error you get when trying to run `docker build -t IO.Swagger .` (or any other `packageName` with upper-case letters):

```
invalid argument "IO.Swagger" for t: Error parsing reference: "IO.Swagger" is not a valid repository/tag: repository name must be lowercase
See 'docker build --help'.
```

2. But even if you manually try to build the project with a correct Docker build tag (**5b)**), it will still fail because the [generated Dockerfile](https://github.com/swagger-api/swagger-codegen/blob/c70a2a7481b34362657546e6ca0f4a41ff1c44b3/modules/swagger-codegen/src/main/resources/aspnetcore/Dockerfile.mustache#L1) uses `FROM microsoft/dotnet:latest` as its base image and there were some breaking changes between `dotnet:1.0.3-sdk` and `dotnet:1.0.4-sdk` (Semver? 🙄 ). [Using `:latest` is very bad practice in Docker-land](https://developers.redhat.com/blog/2016/02/24/10-things-to-avoid-in-docker-containers/); the image associated with it may unexpectedly change in the future (when you've forgotten about it) and then it breaks stuff downstream that depends on an older `:latest`. Use specific version tags instead. For reference, here's the error you get when using `dotnet >= 1.0.4-sdk` or `dotnet == latest`:

```
 ---> Running in f0bfac9cc465
/app/IO.Swagger/IO.Swagger.xproj(7,5): error MSB4019: The imported project "/usr/share/dotnet/sdk/1.0.1/Microsoft/VisualStudio/v14.0/DotNet/Microsoft.DotNet.Props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
The command 'dotnet restore' returned a non-zero code: 1
```

##### Suggest a Fix

Fixes:

1. Do some kind of to-lower-[kebab-case](http://stackoverflow.com/a/12273101) on the `packageName` before sending it to the template. Also, for the Petstore sample, `docker build -t petstore-sample .` is probably a better example, so the Petstore build should probably explicitly set `packageName` to `Petstore.Sample` in a `config.json` instead of relying on `AbstractCSharpCodegen.java`'s `IO.Swagger`.

2.
    a. Fix codegen so that it works with `dotnet >= 1.0.4-sdk`
    b. Or use the latest .NET Core `1.0.x` Docker image which is still compatible with the current setup, and that is `microsoft/dotnet:1.0.3-sdk-projectjson`.

--- a/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AspNetCoreServerCodegen.java
+++ b/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AspNetCoreServerCodegen.java
@@ -86,6 +86,7 @@ public class AspNetCoreServerCodegen extends AbstractCSharpCodegen {
         super.processOpts();
 
         additionalProperties.put("packageGuid", packageGuid);
+        additionalProperties.put("dockerTag", this.packageName.toLowerCase());
 
         apiPackage = packageName + ".Controllers";
         modelPackage = packageName + ".Models";

--- a/modules/swagger-codegen/src/main/resources/aspnetcore/Dockerfile.mustache
+++ b/modules/swagger-codegen/src/main/resources/aspnetcore/Dockerfile.mustache
@@ -1,4 +1,4 @@
-FROM microsoft/dotnet:latest
+FROM microsoft/dotnet:1.0.3-sdk-projectjson
 
 ENV DOTNET_CLI_TELEMETRY_OPTOUT 1
 

--- a/modules/swagger-codegen/src/main/resources/aspnetcore/README.mustache
+++ b/modules/swagger-codegen/src/main/resources/aspnetcore/README.mustache
@@ -22,6 +22,6 @@ build.bat
 
 ```
 cd src/{{packageName}}
-docker build -t {{packageName}} .
-docker run -p 5000:5000 {{packageName}}
+docker build -t {{dockerTag}} .
+docker run -p 5000:5000 {{dockerTag}}
 ```

--- a/samples/server/petstore/aspnetcore/IO.Swagger.sln
+++ b/samples/server/petstore/aspnetcore/IO.Swagger.sln
@@ -7,7 +7,7 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution 
                 global.json = global.json
         EndProjectSection
 EndProject
-Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "IO.Swagger", "src\IO.Swagger\IO.Swagger.xproj", "{795C6B14-1C3E-45F5-AF6F-EE47B197FF1E}"
+Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "IO.Swagger", "src\IO.Swagger\IO.Swagger.xproj", "{3C799344-F285-4669-8FD5-7ED9B795D5C5}"
 EndProject
 Global
         GlobalSection(SolutionConfigurationPlatforms) = preSolution
@@ -15,10 +15,10 @@ Global
                 Release|Any CPU = Release|Any CPU
         EndGlobalSection
         GlobalSection(ProjectConfigurationPlatforms) = postSolution
-                {795C6B14-1C3E-45F5-AF6F-EE47B197FF1E}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-                {795C6B14-1C3E-45F5-AF6F-EE47B197FF1E}.Debug|Any CPU.Build.0 = Debug|Any CPU
-                {795C6B14-1C3E-45F5-AF6F-EE47B197FF1E}.Release|Any CPU.ActiveCfg = Release|Any CPU
-                {795C6B14-1C3E-45F5-AF6F-EE47B197FF1E}.Release|Any CPU.Build.0 = Release|Any CPU
+                {3C799344-F285-4669-8FD5-7ED9B795D5C5}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+                {3C799344-F285-4669-8FD5-7ED9B795D5C5}.Debug|Any CPU.Build.0 = Debug|Any CPU
+                {3C799344-F285-4669-8FD5-7ED9B795D5C5}.Release|Any CPU.ActiveCfg = Release|Any CPU
+                {3C799344-F285-4669-8FD5-7ED9B795D5C5}.Release|Any CPU.Build.0 = Release|Any CPU
         EndGlobalSection
         GlobalSection(SolutionProperties) = preSolution
                 HideSolutionNode = FALSE

--- a/samples/server/petstore/aspnetcore/README.md
+++ b/samples/server/petstore/aspnetcore/README.md
@@ -20,6 +20,6 @@ build.bat
 
 ```
 cd src/IO.Swagger
-docker build -t IO.Swagger .
-docker run -p 5000:5000 IO.Swagger
+docker build -t io.swagger .
+docker run -p 5000:5000 io.swagger
 ```

--- a/samples/server/petstore/aspnetcore/global.json
+++ b/samples/server/petstore/aspnetcore/global.json
@@ -1,7 +1,7 @@
 {
   "projects": [ "src" ],
   "sdk": {
-    "version": "1.0.0-preview3-003171",
+    "version": "1.0.0-preview2-003121",
     "runtime": "coreclr"
   }
 }

--- a/samples/server/petstore/aspnetcore/src/IO.Swagger/Dockerfile
+++ b/samples/server/petstore/aspnetcore/src/IO.Swagger/Dockerfile
@@ -1,4 +1,4 @@
-FROM microsoft/dotnet:latest
+FROM microsoft/dotnet:1.0.3-sdk-projectjson
 
 ENV DOTNET_CLI_TELEMETRY_OPTOUT 1
 

--- a/samples/server/petstore/aspnetcore/src/IO.Swagger/IO.Swagger.xproj
+++ b/samples/server/petstore/aspnetcore/src/IO.Swagger/IO.Swagger.xproj
@@ -6,7 +6,7 @@
     </PropertyGroup>
     <Import Project="$(VSToolsPath)\DotNet\Microsoft.DotNet.Props" Condition="'$(VSToolsPath)' != ''" />
     <PropertyGroup Label="Globals">
-        <ProjectGuid>{795C6B14-1C3E-45F5-AF6F-EE47B197FF1E}</ProjectGuid>
+        <ProjectGuid>{3C799344-F285-4669-8FD5-7ED9B795D5C5}</ProjectGuid>
         <RootNamespace>IO.Swagger</RootNamespace>
         <BaseIntermediateOutputPath Condition="'$(BaseIntermediateOutputPath)'=='' ">.\obj</BaseIntermediateOutputPath>
         <OutputPath Condition="'$(OutputPath)'=='' ">.\bin\</OutputPath>

